quickwit-metastore/src/metastore/single_file_metastore.rs

```diff
@@ -105,23 +126,80 @@ impl Metastore for SingleFileMetastore {
                     .with_error(anyhow::anyhow!("Failed to put metadata set: {:?}", e))
             })?;
 
+        // Update the internal data if the storage is successfully updated.
+        let mut data = self.data.write().await;
+        data.insert(index_uri.clone(), metadata_set.clone());
+
         Ok(())
     }
 
-    async fn delete_index(&self, index_uri: IndexUri) -> MetastoreResult<()> {
+    async fn open_index(&self, index_uri: IndexUri) -> MetastoreResult<()> {
+        // Check for the existence of index.
+        let exists = self.index_exists(index_uri.clone()).await.map_err(|e| {
```

Comment:
Why do we check for existence?
Each call is an extra 70ms of latency.

We can rely on get_all directly.

Reply:
You're absolutely right. I'll fix it.

Reply:
Deleted a redundant existence check at `open_index`.